internal/cephfs/fsjournal.go

```diff
@@ -180,3 +237,145 @@ func reserveVol(ctx context.Context, volOptions *volumeOptions, secret map[strin
 
 	return &vid, nil
 }
+
+// reserveSnap is a helper routine to request a UUID reservation for the CSI SnapName and,
+// to generate the snapshot identifier for the reserved UUID
+func reserveSnap(ctx context.Context, volOptions *volumeOptions, parentSubVolName string, secret map[string]string) (*snapshotIdentifier, error) {
+	var (
+		vid       snapshotIdentifier
+		imageUUID string
+		err       error
+	)
+
+	cr, err := util.NewAdminCredentials(secret)
+	if err != nil {
+		return nil, err
+	}
+	defer cr.DeleteCredentials()
+
+	j, err := snapJournal.Connect(volOptions.Monitors, cr)
+	if err != nil {
+		return nil, err
+	}
+	defer j.Destroy()
+
+	imageUUID, vid.FsSnapshotName, err = j.ReserveName(
+		ctx, volOptions.MetadataPool, util.InvalidPoolID,
+		volOptions.MetadataPool, util.InvalidPoolID, volOptions.RequestName,
+		volOptions.NamePrefix, parentSubVolName, "")
+	if err != nil {
+		return nil, err
+	}
+
+	// generate the snapshot ID to return to the CO system
+	vid.SnapshotID, err = util.GenerateVolID(ctx, volOptions.Monitors, cr, volOptions.FscID,
+		"", volOptions.ClusterID, imageUUID, volIDVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	klog.V(4).Infof(util.Log(ctx, "Generated Snapshot ID (%s) for request name (%s)"),
```

Comment:
Please refer to https://github.com/ceph/ceph-csi/pull/1214 for updated logging mechanism.